### DISCUSSION TITLE
Can not reproduce cdm-s3 example

### DISCUSSION THREAD

**Discussion by valpo-syncretis** [Unidata/netcdf-java#1056] [Questions and Answers] [answered]
Hello!
I'm trying to reproduce example from https://docs.unidata.ucar.edu/netcdf-java/current/userguide/dataset_urls.html#object-stores.

String host = "storage.googleapis.com";
String bucket = "gcp-public-data-goes-16";
String key = "ABI-L1b-RadC/2017/242/00/OR_ABI-L1b-RadC-M3C01_G16_s20172420002168_e20172420004540_c20172420004583.nc";
String cdmS3Uri = "cdms3://" + host + "/" + bucket + "?" + key;
NetcdfFile file = NetcdfFiles.open(cdmS3Uri);

It always throws java.io.FileNotFoundException: cdms3:\storage.googleapis.com\gcp-public-data-goes-16?ABI-L1b-RadC\2017\242\00\OR_ABI-L1b-RadC-M3C01_G16_s20172420002168_e20172420004540_c20172420004583.nc (The filename, directory name, or volume label syntax is incorrect).

Tthese are my depemdencies from .pom:
		<dependency>
			<groupId>edu.ucar</groupId>
			<artifactId>cdm-s3</artifactId>
			<version>5.3.0</version>
		</dependency>
		<dependency>
			<groupId>edu.ucar</groupId>
			<artifactId>netcdf-java</artifactId>
			<version>5.3.3</version>
		</dependency>

Can you please tell me what I'm doing wrong? I have access to the cloud, since it's public. I tried other formats for the url but no progress.  May be I should add something else to .pom? Many thanks for the answer.

**Comment by lesserwhirls** [comment, marked as answer]
Can you try the example using the following artifacts:

edu.ucar:cdm-core:5.3.3
edu.ucar:cdm-s3:5.3.3

The latest version is 5.5.3, so I would also suggest upgrading to that version if at all possible.

**Reply by valpo-syncretis** [reply]
Thank you very much for the answer. It works! It is always difficult for me to find out what artifacts are needed.

**Reply by valpo-syncretis** [reply]
One more question regarding opening files via cdms3. Is it possible to open .pgrb2 file with cdms3? I can open .pgrb2 file from my local computer using NetcdfFiles.open() and adding dependency to grib artifact. But I can not open .pgrb2 file from object storage.

**Comment by rschmunk** [comment]
I had no trouble remote loading the address

cdms3://storage.googleapis.com/gcp-public-data-goes-16?ABI-L1b-RadC/2017/242/00/OR_ABI-L1b-RadC-M3C01_G16_s20172420002168_e20172420004540_c20172420004583.nc

Whereas OP is citing an address with backslashes rather than forward slashes.

Edit: I see why that specific date/time for a sample dataset was chosen.
